### PR TITLE
Add addresses(access IP) into output list

### LINKED ISSUE #950: Add Addresses  into output list
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]

per https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/331
we want to floating ip (access ip) to be showable in the 'get' command
this will require changes in cluster-api

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by jichenjc**
at some siutation some provider such as openstack need to know the
access ip to help the problem debug, this PR adds the info.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #950 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: config/crds/cluster_v1alpha1_machine.yaml [hunk 1]
```diff
@@ -15,6 +15,10 @@ spec:
     description: Machine status such as Terminating/Pending/Running/Failed etc
     name: Phase
     type: string
+  - JSONPath: .status.addresses
+    description: Access IP(s) of the machine
+    name: Addresses
+    type: string
```
Comment (ncdc) [inline config/crds/cluster_v1alpha1_machine.yaml]: Will this print out correctly as a string, given the type is `[]corev1.NodeAddress`?

**Comment by detiber** [comment]
This seems to mainly be a workaround for the fact that we do not have remote node references. Is there a use case where attaching IP addresses directly to the Machine would be preferred over a reference to a Node that has the listed IP addresses?

The one case that I could potentially think of would be if the Machine fails to progress to the state of a booted Node, but it should be possible for a user to connect the dots based on the providerID.

**Comment by jichenjc** [comment]
@hchenxa this is for the https://github.com/kubernetes-sigs/cluster-api-provider-openstack/issues/331 , I think @detiber provided a good comment on we can get the Node address from the `get node`  command? I didn't realize that so do we need to consider use the address from node directly after machine provisioned successfully?  The failed case mentioned above by @detiber seems reasonable but I don't know whether it's real needed

**Comment by jichenjc** [comment]
FYI @chrigl @gyliu513

**Comment by vincepri** [comment]
@jichenjc What's the status of this PR?

**Comment by jichenjc** [comment]
I will close this as node address seems works for me ,will reopen if needed

**Pull request closed without merging**